source/OrderedTypes/InductiveTypesfromSmallGeneratedLattice.lagda

```diff
@@ -0,0 +1,2209 @@
+                              (t : T is 𝓥 small)
+                              where
+ 
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ _≤_ = order-of L
+
+ ⋁_ : Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+ ⋁_ = join-for L
+
+ T' : 𝓥  ̇
+ T' = (resized T) t
+
+ T'-≃-T : T' ≃ T
+ T'-≃-T = resizing-condition t
+
+ T'-to-T : T' → T
+ T'-to-T = ⌜ T'-≃-T ⌝
+
+ T'-to-T-is-equiv : is-equiv T'-to-T
+ T'-to-T-is-equiv = ⌜ T'-≃-T ⌝-is-equiv
+
+ T-to-T' : T → T'
+ T-to-T' =  ⌜ T'-≃-T ⌝⁻¹
+
+ T'-to-T-has-section : has-section T'-to-T
+ T'-to-T-has-section = equivs-have-sections T'-to-T T'-to-T-is-equiv
+
+ T'-to-T-is-section : is-section T'-to-T
+ T'-to-T-is-section = equivs-are-sections T'-to-T T'-to-T-is-equiv
+
+ section-T'-to-T : T'-to-T ∘ T-to-T' ∼ id
+ section-T'-to-T = section-equation T'-to-T T'-to-T-has-section
+
+ retraction-T'-to-T : T-to-T' ∘ T'-to-T ∼ id
+ retraction-T'-to-T = inverses-are-retractions' T'-≃-T
+
+ T'-inclusion : T' → ⟨ L ⟩
+ T'-inclusion = m ∘ T'-to-T
+
+ s : ⟨ L ⟩
+ s = ⋁ (T' , T'-inclusion)
+
+ open Joins _≤_
+
+ sup-of-small-fam-is-lub : (s is-lub-of (T , m)) holds
+ sup-of-small-fam-is-lub = (s-upper-bound , s-least-upper-bound)
+  where
+   s-upper-bound : (s is-an-upper-bound-of (T , m)) holds
+   s-upper-bound t = t-≤-s
+    where
+     t-≤-s : (m t ≤ s) holds 
+     t-≤-s = transport (λ z → (m z ≤ s) holds) (section-T'-to-T t)
+                       ((is-an-upper-bound-for L of
+                        (T' , T'-inclusion)) (T-to-T' t))
+   s-least-upper-bound : ((u , _) : upper-bound (T , m)) → (s ≤ u) holds
+   s-least-upper-bound (u , is-upbnd-T) = s-≤-u
+    where
+     s-≤-u : (s ≤ u) holds
+     s-≤-u = pr₂ (is-lub-for L (T' , T'-inclusion))
+                 ((u , λ i → is-upbnd-T (T'-to-T i)))
+
+\end{code}
+
+We now show that reindexing across an equivalence does not change the join.
+
+\begin{code}
+
+module equivalent-families-have-same-join {𝓤 𝓦 𝓥 𝓣 𝓣' : Universe}
+                                          (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                                          (T : 𝓣  ̇)
+                                          (T' : 𝓣'  ̇)
+                                          (e : T' ≃ T)
+                                          (m : T → ⟨ L ⟩)
+                                          where
+
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ _≤_ = order-of L
+
+ ⋁_ : Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+ ⋁_ = join-for L
+
+ open Joins _≤_
+
+ reindexing-along-equiv-＝-sup : (s s' : ⟨ L ⟩)
+                               → (s is-lub-of (T , m)) holds
+                               → (s' is-lub-of (T' , m ∘ ⌜ e ⌝ )) holds
+                               → s ＝ s'
+ reindexing-along-equiv-＝-sup s s' (is-upbnd , is-least-upbnd)
+                        (is-upbnd' , is-least-upbnd') =
+   is-antisymmetric-for L s-≤-s' s'-≤-s
+  where
+   s-≤-s' : (s ≤ s') holds
+   s-≤-s' = is-least-upbnd (s' , λ t → transport (λ z → (z ≤ s') holds)
+                           (＝₁ t) (is-upbnd' (⌜ e ⌝⁻¹ t)))
+    where
+     ＝₁ : (t : T) → m (⌜ e ⌝ (⌜ e ⌝⁻¹ t)) ＝ m t
```

Comment:
This notation suggests that you are defining another notion of equality, so it's not a very good choice.